src/Native/Runtime/arm64/Interlocked.S

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+//
+
+#include <unixasmmacros.inc>
+
+// WARNING: Code in EHHelpers.cpp makes assumptions about this helper, in particular:
+// - Function "InWriteBarrierHelper" assumes an AV due to passed in null pointer will happen at RhpLockCmpXchg32AVLocation
+// - Function "UnwindWriteBarrierToCaller" assumes the stack contains just the pushed return address
+// x0 = destination address
+// w1 = value
+// w2 = comparand
+LEAF_ENTRY RhpLockCmpXchg32, _TEXT
+    mov     x8, x0          // Save value of x0 into x8 as x0 is used for the return value
+ALTERNATE_ENTRY RhpLockCmpXchg32AVLocation
+1: // loop
+    ldaxr   w0, [x8]        // w0 = *x8
+    cmp     w0, w2 
+    bne     2f              // if (w0 != w2) goto exit
+    stlxr   w9, w1, [x8]    // if (w0 == w2) { try *x8 = w1 and goto loop if failed or goto exit }
+    cbnz    w9, 1b
+2: // exit
```

Comment:
You are right. It used to be necessary for ARM v7. I have not realized that it is no longer necessary for ARM v8.